src/object/object_representation.c

```diff
@@ -1352,23 +1354,91 @@ or_put_varchar_internal (OR_BUF * buf, char *string, int charlen, int align)
   else
     {
       rc = or_put_byte (buf, 0xFF);
-
-      if (rc == NO_ERROR)
-	{
-	  OR_PUT_INT (&net_charlen, charlen);
-	  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
-	}
+      compressable = 1;
     }
   if (rc != NO_ERROR)
     {
       return rc;
     }
 
-  /* store the string bytes */
-  rc = or_put_data (buf, string, charlen);
-  if (rc != NO_ERROR)
+  if (compressable == 1)
     {
-      return rc;
+      /* Alloc memory */
+      wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
+      if (wrkmem == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) LZO1X_1_MEM_COMPRESS);
+	  goto cleanup;
+	}
+
+      /* Alloc memory for the compressed string */
+      compressed_string = (char *) malloc ((charlen + (charlen / 16) + 64 + 3) * sizeof (char));
```

Comment:
Memory allocation success needs to be checked.
Future optimization : use a preallocated buffer (per thread_entry) to store compressed string (when size is not large), to avoid memory allocation overhead.